ws-butler.el

```diff
@@ -136,21 +145,24 @@ Also see `require-final-newline'."
 If `indent-tabs-mode' is nil, then tabs in indentation is
 replaced by spaces."
   (interactive "*r")
-  (ws-butler-with-save
-   (narrow-to-region beg end)
-   ;;  _much slower would be:       (replace-regexp "[ \t]+$" "")
-   (goto-char (point-min))
-   (while (not (eobp))
-     ;; convert leading tabs to spaces
-     (unless indent-tabs-mode
-       (let ((eol (point-at-eol)))
-         (skip-chars-forward " " (point-at-eol))
-         (when (eq (char-after) ?\t )
-           (untabify (point) (progn (skip-chars-forward " \t" (point-at-eol))
-                                    (point))))))
-     (end-of-line)
-     (delete-horizontal-space)
-     (forward-line 1)))
+  ;; check if user wants to skip the region
+  (unless (and (fboundp 'ws-butler-skip-strip-function)
```

Comment:
better to decide cleanup before entering clean function, i.e. https://github.com/lewang/ws-butler/blob/a998a23ee8713808ac1fe3d1523ea1861be4da57/ws-butler.el#L203